programs/uxd/src/instructions/credix_lp/redeem_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,494 @@
+        ],
+        bump = depository.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+        has_one = collateral_mint @UxdError::InvalidCollateralMint,
+        has_one = depository_collateral @UxdError::InvalidDepositoryCollateral,
+        has_one = depository_shares @UxdError::InvalidDepositoryShares,
+        has_one = credix_program_state @UxdError::InvalidCredixProgramState,
+        has_one = credix_global_market_state @UxdError::InvalidCredixGlobalMarketState,
+        has_one = credix_signing_authority @UxdError::InvalidCredixSigningAuthority,
+        has_one = credix_liquidity_collateral @UxdError::InvalidCredixLiquidityCollateral,
+        has_one = credix_shares_mint @UxdError::InvalidCredixSharesMint,
+    )]
+    pub depository: AccountLoader<'info, CredixLpDepository>,
+
+    /// #5
+    #[account(mut)]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+
+    /// #6
+    pub collateral_mint: Box<Account<'info, Mint>>,
+
+    /// #7
+    #[account(
+        mut,
+        constraint = user_redeemable.owner == user.key() @UxdError::InvalidOwner,
+        constraint = user_redeemable.mint == redeemable_mint.key() @UxdError::InvalidRedeemableMint,
+    )]
+    pub user_redeemable: Box<Account<'info, TokenAccount>>,
+
+    /// #8
+    #[account(
+        mut,
+        constraint = user_collateral.owner == user.key() @UxdError::InvalidOwner,
+        constraint = user_collateral.mint == collateral_mint.key() @UxdError::InvalidCollateralMint
+    )]
+    pub user_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #9
+    #[account(mut)]
+    pub depository_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #10
+    #[account(mut)]
+    pub depository_shares: Box<Account<'info, TokenAccount>>,
+
+    /// #11
+    #[account(
+        has_one = credix_multisig_key @UxdError::InvalidCredixMultisigKey,
+    )]
+    pub credix_program_state: Box<Account<'info, credix_client::ProgramState>>,
+
+    /// #12
+    #[account(
+        mut,
+        constraint = credix_global_market_state.treasury_pool_token_account == credix_treasury_collateral.key() @UxdError::InvalidCredixTreasuryCollateral,
+    )]
+    pub credix_global_market_state: Box<Account<'info, credix_client::GlobalMarketState>>,
+
+    /// #13
+    /// CHECK: unused by us, checked by credix
+    pub credix_signing_authority: AccountInfo<'info>,
+
+    /// #14
+    #[account(mut)]
+    pub credix_liquidity_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #15
+    #[account(mut)]
+    pub credix_shares_mint: Box<Account<'info, Mint>>,
+
+    /// #16
+    #[account(
+        mut,
+        constraint = credix_pass.user == depository.key() @UxdError::InvalidCredixPass,
+        //constraint = credix_pass.disable_withdrawal_fee == true @UxdError::InvalidCredixPassNoFees,
+    )]
+    pub credix_pass: Box<Account<'info, credix_client::CredixPass>>,
+
+    /// #17
+    #[account(
+        mut,
+        token::mint = collateral_mint,
```

Comment:
mint checking here is also optional right?
the `credix_global_market_state` has to be matched with the depository and this `credix_treasury_collateral` has to be matched with the `credix_global_market_state.treasury_pool_token_account`

Reply:
yep adding this just in case